capstone-app/src/app/modules/maps/map/map.component.ts

```diff
@@ -1,13 +1,80 @@
 /// <reference types="googlemaps" />
-import { Component } from '@angular/core';
+import { Component,  ViewChild, ElementRef } from '@angular/core';
+import { MapInfoWindow, MapMarker } from '@angular/google-maps';
 
 @Component({
   selector: 'app-map',
   templateUrl: './map.component.html',
   styleUrls: ['./map.component.scss']
 })
+
 export class MapComponent {
-  // these coordinates are for a location in Miami
-  location = new google.maps.LatLng(26.011760, -80.139050);
+  @ViewChild('searchBar') searchBar: ElementRef;
+  @ViewChild(MapInfoWindow) infoWindow: MapInfoWindow;
+
+  location!: google.maps.LatLng;
+  zoom = 13;
+  position!: google.maps.LatLngLiteral;
+  clickLocation: google.maps.LatLng[] = [];
+  markers: google.maps.LatLng[] = [];
+
+  ngAfterViewInit() {
+    setTimeout(() => {
+     const input = this.searchBar.nativeElement;
+     const autoComplete = new google.maps.places.Autocomplete(input,
+       {fields: ['geometry', 'name'], types: ['(cities)']});
+     autoComplete.addListener('place_changed', () => {
+       this.location = autoComplete.getPlace().geometry.location;
+     });
+    });
+  }
+
+  // update location using search button instead of input bar 
+  searchButton(barcation: String) {
+    const input = this.searchBar.nativeElement;
+    const autoComplete = new google.maps.places.Autocomplete(input,
+      {fields: ['geometry', 'name'], types: ['(cities)']});
+    autoComplete.addListener('place_changed', () => {
+      this.location = autoComplete.getPlace().geometry.location;
+    });
+  }
+
+  getCurrentOrSetLocation() {
+    if (!this.location) {
+      return new google.maps.LatLng(this.position);
+    }
+    return this.location;
+  }
+
+  ngOnInit() {
+    this.getCurrentLocation();
+    this.getCurrentOrSetLocation();
+  }
```

Comment:
nit: let's move ngOnInit to be above ngAfterViewInit. It is helpful to keep the Angular lifecycle hook implementations above the component-specific methods